graphql/src/main/java/com/scalar/db/graphql/instrumentation/validation/AbortFieldValidation.java

```diff
@@ -0,0 +1,77 @@
+package com.scalar.db.graphql.instrumentation.validation;
+
+import com.google.common.collect.ImmutableList;
+import com.scalar.db.graphql.schema.Constants;
+import graphql.ErrorType;
+import graphql.GraphQLError;
+import graphql.GraphqlErrorBuilder;
+import graphql.execution.instrumentation.fieldvalidation.FieldValidation;
+import graphql.execution.instrumentation.fieldvalidation.FieldValidationEnvironment;
+import graphql.language.Argument;
+import graphql.language.BooleanValue;
+import graphql.language.Directive;
+import graphql.language.Field;
+import graphql.language.NullValue;
+import graphql.language.OperationDefinition;
+import graphql.language.Selection;
+import graphql.language.SourceLocation;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+public class AbortFieldValidation implements FieldValidation {
+  @SuppressWarnings("rawtypes")
+  @Override
+  public List<GraphQLError> validateFields(FieldValidationEnvironment validationEnvironment) {
+    OperationDefinition operationDefinition =
+        validationEnvironment.getExecutionContext().getOperationDefinition();
+    List<Selection> selections = operationDefinition.getSelectionSet().getSelections();
+    Optional<Selection> abortField =
+        selections.stream()
+            .filter(selection -> "abort".equals(((Field) selection).getName()))
+            .findFirst();
+    if (!abortField.isPresent()) {
+      return Collections.emptyList();
+    }
+
+    ImmutableList.Builder<GraphQLError> errors = ImmutableList.builder();
+    List<Directive> transactionDirectives =
+        operationDefinition.getDirectives(Constants.TRANSACTION_DIRECTIVE_NAME);
+    SourceLocation location = abortField.get().getSourceLocation();
+    if (transactionDirectives.isEmpty()) {
+      errors.add(
+          createError(
+              "transaction directive with txId is required to abort transaction", location));
+    } else {
+      Directive directive = transactionDirectives.get(0); // @transaction is not repeatable
```

Comment:
What do you mean by this? `@transaction is not repeatable`

Reply:
Directives can be repeatable if defined so, but we don't define `@transaction` as such. So, if the list is not empty, we can only pay attention to the first element (`get(0)`).

Reply:
I didn't get the meaning of `repeatable` here, but after some investigation, I got it. Thank you!